### PR TITLE
Implement budget threshold notifications for expenses

### LINKED ISSUE #690: [Backend] Send budget threshold notifications to project users on expense create/update
When an expense's actual amount crosses a budget threshold, all accepted users in the project should receive an in-app notification. Two thresholds are tracked:
- 80% — early warning that the budget is nearly exhausted
- 100% — budget has been exceeded
If a single create/update causes the spend to jump from below 80% directly past 100%, only the 100% notification should fire (skip the 80% one).

Implementation notes:

- Logic lives in ExpenseService.CreateAsync and UpdateAsync, called after CommitAsync
- A private CheckBudgetAlertsAsync(Guid projectId, Expense expense) method:
  1. Guards on expense.Budget == 0 to avoid division by zero
  2. Computes percentage = expense.Amount / expense.Budget * 100
  3. If >= 100% → sends overflow notification to all users
  4. else if >= 80% → sends warning notification to all users

- Project users are fetched from IUnitOfWork.UserProjectRepository (only Accepted == true)
- Notifications are dispatched via INotificationService.CreateNotificationAsync with NotificationCategory.Finance
- New i18n translation keys are needed for both message codes

Acceptance criteria:

- [ ] Creating an expense that immediately exceeds 100% of budget notifies all project users with the overflow message only
- [ ] Updating an expense that crosses the 80% threshold notifies all project users with the warning message
- [ ] No duplicate or double notifications when jumping from <80% to >100% in one operation
- [ ] Budget = 0 does not cause errors (no notification sent)
- [ ] Unit tests cover all threshold edge cases and the jump-past-80% scenario
- [ ] Integration tests verify dis

patch through the full expense service flow

### PR DESCRIPTION
**Pull request by Kiri-03**
### Description
This PR implements the budget alert notification system as requested in issue #690.

### Changes
- Implemented `CheckBudgetAlertsAsync` in `ExpenseService` to monitor budget thresholds.
- - Added logic for 80% (Warning) and 100% (Overflow) notifications.
- - Ensured that if an expense jumps directly past 100%, only the overflow notification is sent.
- - Integrated notifications in both `CreateAsync` and `UpdateAsync`.
- - Added missing i18n keys for `BUDGET_WARNING` and `BUDGET_OVERFLOW` in English and Portuguese.
- - Added comprehensive unit tests in `ExpenseServiceTests.cs` covering all scenarios.
### Validation
- Verified all unit tests pass in a containerized environment (Docker).
- - Confirmed correct calculation of percentages using decimal precision.
Fixes #690

### CODE REVIEW

**Comment by Kiri-03** [comment]
@FelipePSoares, this PR is ready for review. It implements the budget alert system for issue #690.

**Comment by FelipePSoares** [comment]
Hi @Kiri-03,

So fast, thank you so much,  I'll review it as soon as possible.

File: EasyFinance.Application/Features/ExpenseService/ExpenseService.cs [hunk 1]
```diff
@@ -341,5 +355,42 @@ public async Task<AppResponse> RemoveLinkAsync(User user)
...
+            var userProjects = await unitOfWork.UserProjectRepository
+                .NoTrackable()
+                .Include(up => up.Project)
+                .Include(up => up.User)
+                .ToListAsync();
+
+            var users = userProjects
+                .Where(up => up.Project?.Id == projectId && up.Accepted)
+                .Select(up => up.User)
+                .ToList();
```
Comment (FelipePSoares) [inline EasyFinance.Application/Features/ExpenseService/ExpenseService.cs]: Hi @Kiri-03 ,

I suggest make the where and even the select before the ToList to avoid get all the data from database to filter after.

File: EasyFinance.Application/Features/ExpenseService/ExpenseService.cs [hunk 2]
```diff
@@ -341,5 +355,42 @@ public async Task<AppResponse> RemoveLinkAsync(User user)
...
+
+            foreach (var user in users)
+            {
+                await notificationService.CreateNotificationAsync(new NotificationRequestDTO
+                {
+                    User = user,
+                    Type = NotificationType.Information,
+                    CodeMessage = messageCode,
+                    Category = NotificationCategory.Finance
+                });
```
Comment (FelipePSoares) [inline EasyFinance.Application/Features/ExpenseService/ExpenseService.cs]: I suggest add push, in app and web push limitation to avoid notify user by email.

```suggestion
                await notificationService.CreateNotificationAsync(new NotificationRequestDTO
                {
                    User = user,
                    Type = NotificationType.Information,
                    CodeMessage = messageCode,
                    Category = NotificationCategory.Finance,
                    LimitNotificationChannels = NotificationChannels.Push | NotificationChannels.InApp | NotificationChannels.WebPush
                });
```

**Comment by Kiri-03** [comment]
Thanks, @FelipePSoares ! I just pushed a fix for the integration tests that were failing in the CI due to the new dependency injection. Everything should be green now once the workflow runs.

**Review by FelipePSoares — CHANGES_REQUESTED** [review]
Hi @Kiri-03,

I got some suggestions of improvements in your code.

besides the points below I would like to ask you to take a look inside `WebPushService.ResolveNotificationBody` to resolve these new messages there, to do it you will need to add the messages that you add in the `messages.en.json` and `messages.pt.json` in the `NotificationMessages.resx` and `NotificationMessages.pt.resx` as well. 😉

Thank you in advance.

**Comment by Kiri-03** [comment]
Hi @FelipePSoares,

Thanks for the feedback! I've updated the PR with the optimized DB query and limited the notification channels as suggested. Ready for another look! 🚀

**Comment by FelipePSoares** [comment]
Hi @Kiri-03, 

Thank you so much, really nice work. 

I'll just ask three more small things and I swear that is the last things. Ahaha

- add a try catch to just log any error in the `CheckBudgetAlertsAsync`, to avoid throw an exception in case of any exception in this notification process. 
- add the same messages that you added in the front-end in the notifications.resx inside the infrastructure project.
- resolve these code messages in the WebPushService.ResolveNotificationBody using the NotificationMessages.resx.

After this I'll approve the pull request. I swear. Hahaha

Thank you so much for your help.

**Comment by Kiri-03** [comment]
Hi @FelipePSoares! Done! ✅

I've added three more small things hahah, the try-catch block with logging to CheckBudgetAlertsAsync, updated both NotificationMessages.resx files in the Infrastructure project, and updated WebPushService to resolve code messages using the ResourceManager. Everything should be ready now! 🚀

**Review by FelipePSoares — APPROVED** [review]
Uuuhhhuuu nice work, 

Thank you again, feel free to take another one if you liked to work in this project or even suggest new tasks. 

I wish you all the best.
